.dotnet/src/Custom/Chat/ChatClient.cs

```diff
@@ -65,6 +66,7 @@ protected internal ChatClient(ClientPipeline pipeline, string model, Uri endpoin
         _pipeline = pipeline;
         _endpoint = endpoint;
         _telemetry = new OpenTelemetrySource(model, endpoint);
+        _exceptionFactory = new OpenAIExceptionFactory();
```

Comment:
Customization layer sets this in the client constructor, or generated code sets it to ClientResultExceptionFactory.Default.